drivers/mlx90393/mlx90393.c

```diff
@@ -0,0 +1,625 @@
+        xyz_resolution_value_mask)) != 0) {
+        _release(dev);
+        return error;
+    }
+    /* burst more */
+    if (DEV_MODE == MLX90393_MODE_BURST) {
+        /* set burst data rate */
+        if ((error = _write_register_bits(dev, MLX90393_REG_CONF1, MLX90393_MASK_BDR,
+            DEV_ODR)) != 0) {
+            _release(dev);
+            return error;
+        }
+        /* start burst mode */
+        if ((error = _write_byte(dev, MLX90393_COMMAND_SB)) != 0) {
+            _release(dev);
+            return error;
+        }
+        if ((error = _check_status_byte(dev)) != 0) {
+            _release(dev);
+            return error;
+        }
+    }
+    /* wake up on change mode */
+    else if (DEV_MODE == MLX90393_MODE_WOC_ABSOLUTE || DEV_MODE == MLX90393_MODE_WOC_RELATIVE) {
+        if (!gpio_is_valid(DEV_INT_PIN)) {
+            _release(dev);
+            DEBUG("[mlx90393] error: No valid interrupt pin passed in params\n");
+            return -EINVAL;
+        }
+        /* set absolute or relative wake up on change mode */
+        int woc_mode = 0;
+        if (DEV_MODE == MLX90393_MODE_WOC_RELATIVE) {
+            woc_mode = 1;
+        }
+        woc_mode <<= MLX90393_SHIFT_WOC_MODE;
+        if ((error = _write_register_bits(dev, MLX90393_REG_CONF1, MLX90393_MASK_WOC_DIFF,
+            woc_mode)) != 0) {
+            _release(dev);
+            return error;
+        }
+        /* set burst data rate */
+        if ((error = _write_register_bits(dev, MLX90393_REG_CONF1, MLX90393_MASK_BDR,
+            DEV_ODR)) != 0) {
+            _release(dev);
+            return error;
+        }
+        /* set thresholds */
+        int gain = _get_gain_factor(DEV_GAIN);
+        uint16_t raw_xy_threshold = (1000 * dev->params->threshold.xy / (MLX90393_XY_SENS
+            * (1 << DEV_RESOLUTION) * gain)) * 100;
+        uint16_t raw_z_threshold = (1000 * dev->params->threshold.z / (MLX90393_Z_SENS
+            * (1 << DEV_RESOLUTION) * gain)) * 100;
+        uint16_t raw_temp_threshold = dev->params->threshold.temp * MLX90393_TEMP_RESOLUTION
+            / 1000;
+
+        if ((error = _write_register(dev, MLX90393_REG_WOXY_THRESHOLD, raw_xy_threshold))) {
+            _release(dev);
+            return error;
+        }
+        if ((error = _write_register(dev, MLX90393_REG_WOZ_THRESHOLD, raw_z_threshold))) {
+            _release(dev);
+            return error;
+        }
+        if ((error = _write_register(dev, MLX90393_REG_WOT_THRESHOLD, raw_temp_threshold))) {
+            _release(dev);
+            return error;
+        }
+        /* start wake up on change mode */
+        if ((error = _write_byte(dev, MLX90393_COMMAND_SW)) != 0) {
+            _release(dev);
+            return error;
+        }
+        if ((error = _check_status_byte(dev)) != 0) {
+            _release(dev);
+            return error;
+        }
+    }
+    if (DEV_MODE == MLX90393_MODE_SINGLE_MEASUREMENT && !gpio_is_valid(DEV_INT_PIN)) {
```

Comment:
Siehe Anmerkung zu Pseudomodule `mlx90393_int`:
```suggestion
    if (!ISUSED(MODULE_MLX90393) && (DEV_MODE == MLX90393_MODE_SINGLE_MEASUREMENT)) {
```